sql/ddl_internal.sql

```diff
@@ -119,10 +119,18 @@ BEGIN
 
     IF num_slices IS NULL THEN
         -- Open dimension
-        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
+        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'DATE', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
             RAISE EXCEPTION 'illegal type for column "%": %', column_name, column_type
             USING ERRCODE = 'IO102';
         END IF;
+        IF column_type = 'DATE'::regtype AND interval_length < _timescaledb_internal.interval_to_usec('1 day') THEN
+            RAISE EXCEPTION 'The interval for a hypertable partitioned by a date column must be greater than a day.'
+            USING ERRCODE = 'IO102';
+        END IF;
+        IF column_type = 'DATE'::regtype AND interval_length % _timescaledb_internal.interval_to_usec('1 day') != 0 THEN
+            RAISE EXCEPTION 'The interval for a hypertable partitioned by a date column must be in intervals of a day.'
```

Comment:
The reason I had them separate is to give different error msgs. What should this error msg be for the combined version?